src/models/mod.rs

```diff
@@ -1,36 +1,61 @@
 use log::info;
 
-use crate::Error;
+use crate::{models, Error};
 
 pub mod db;
 #[cfg(passkey)]
 pub mod keys;
+mod migrations;
 pub mod passwords;
 pub mod users;
 
+static CREATE_USERS_TABLE: &str = "CREATE TABLE IF NOT EXISTS users (
+        id TEXT PRIMARY KEY,
+        username TEXT UNIQUE,
+        hash TEXT,
+        salt TEXT
+        );";
+static CREATE_KEYS_TABLE: &str = "CREATE TABLE IF NOT EXISTS keys (
+        id INT PRIMARY KEY,
+        userid TEXT,
+        key TEXT
+        );";
+
+// this array should not be changed, only added to
+static MIGRATIONS: [&str; 4] = [
+    CREATE_USERS_TABLE,
+    CREATE_KEYS_TABLE,
+    "CREATE TABLE IF NOT EXISTS migrations (
+        id INT PRIMARY KEY
+    );",
```

Comment:
should probably add a date column, "date_executed", and a string for the query executed maybe?